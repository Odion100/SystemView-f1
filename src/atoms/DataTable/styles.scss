.data-table {
  width: 100%;
  margin: 5px 0;
  background-color: #ffffff;
  border-collapse: collapse;
  color: #000000;
  & td,
  & th {
    border-width: 1px;
    border-color: #bfbfbe;
    border-style: solid;
    padding: 5px;
    font-weight: 300;
  }
  & th {
    font-weight: bold;
    font-size: 15px;
    color: #3167bf;
    padding: 6px;
  }
  & thead {
    background-color: #dadada;
  }
}
