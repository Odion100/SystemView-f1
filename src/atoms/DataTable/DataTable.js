import React from "react";
import "./styles.scss";

const DataTable = ({ headers = [], table = [], tableClassName }) => {
  return (
    <table className={`data-table ${tableClassName}`}>
      <thead>
        <tr>
          {headers.map(({ name }, i) => (
            <th key={i}>{name}</th>
          ))}
        </tr>
      </thead>
      <tbody>
        {table.map((row, i) => (
          <tr key={i}>
            {row.map((value, i) => (
              <td key={i}>{value}</td>
            ))}
          </tr>
        ))}
      </tbody>
    </table>
  );
};

export default DataTable;
