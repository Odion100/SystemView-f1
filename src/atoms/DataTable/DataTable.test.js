/* eslint-disable jest/valid-expect */
import React from "react";
import ReactDom from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import DataTable from "./DataTable";

afterEach(cleanup);

describe("<DataTable/>", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(<DataTable />, div);
  });

  it("renders an table element with text and class button", () => {
    const { getByRole } = render(<DataTable />);
    expect(getByRole("table")).toHaveClass("data-table");
  });

  it("renders content correctly", () => {
    const headers = [{ name: "column 1" }, { name: "column 2" }];
    const data = [
      ["r1c1", "r1c2"],
      ["r2c1", "r2c2"],
    ];
    const { getByRole, getAllByRole } = render(
      <DataTable tableClassName={"test"} headers={headers} table={data} />
    );

    expect(getByRole("table")).toHaveClass("data-table");
    expect(getByRole("table")).toHaveClass("test");
    expect(getAllByRole("columnheader")).toHaveLength(headers.length);
    expect(getAllByRole("columnheader")[0]).toHaveTextContent(headers[0].name);
    expect(getAllByRole("columnheader")[1]).toHaveTextContent(headers[1].name);
    expect(getAllByRole("cell")).toHaveLength(data.length * headers.length);
    expect(getAllByRole("cell")[0]).toHaveTextContent(data[0][0]);
    expect(getAllByRole("cell")[1]).toHaveTextContent(data[0][1]);
    expect(getAllByRole("cell")[2]).toHaveTextContent(data[1][0]);
    expect(getAllByRole("cell")[3]).toHaveTextContent(data[1][1]);
  });

  it("should match snapshot", () => {
    const headers = [{ name: "column 1" }, { name: "column 2" }];
    const data = [
      ["r1c1", "r1c2"],
      ["r2c1", "r2c2"],
    ];
    const tree = renderer
      .create(<DataTable tableClassName={"test"} headers={headers} table={data} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
