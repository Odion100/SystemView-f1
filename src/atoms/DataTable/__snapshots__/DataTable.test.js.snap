// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataTable/> should match snapshot 1`] = `
<table
  className="data-table test"
>
  <thead>
    <tr>
      <th>
        column 1
      </th>
      <th>
        column 2
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        r1c1
      </td>
      <td>
        r1c2
      </td>
    </tr>
    <tr>
      <td>
        r2c1
      </td>
      <td>
        r2c2
      </td>
    </tr>
  </tbody>
</table>
`;
